plugins/module_utils/aci.py

```diff
@@ -1358,3 +1293,58 @@ def dump_json(self):
             if(output_path is not None):
                 with open(output_path, "a") as output_file:
                     json.dump([mo], output_file)
+
+    def call(self, method):
+        if method == 'GET':
+            call_path = self.path + self.filter_string
+            call_url = self.url + self.filter_string
+            data = None
+        elif method == 'POST':
+            call_path = self.path
+            call_url = self.url
+            data = json.dumps(self.config)
+        elif method == 'DELETE':
+            call_path = self.path
+            call_url = self.url
+            data = None
+        resp = None
+        if self.params.get('private_key'):
+            self.cert_auth(method=method, path=call_path, payload=data)
+
+        if self.module._socket_path:
+            conn = Connection(self.module._socket_path)
+            conn.get_auth(self.get_auth, self.params.get('host'), self.params.get('username'), self.params.get('password'))
```

Comment:
Why not pass it to the connection init directly? I would in fact pass the whole "self" object to the Connect() constructor and then you are free on the other side to use it.

Reply:
There are two Connection classes. This particular class has only one argument which is the socket path.